phalcon/Cache.zep

```diff
@@ -205,9 +205,9 @@ class Cache implements CacheInterface
     {
         let key = (string) key;
 
-        if preg_match("/[^A-Za-z0-9-_.]/", key) {
+        if preg_match("/[^A-Za-z0-9-_:\\\.]/", key) {
```

Comment:
Need to check if all adapters support that type of keys.

Reply:
Yea, i was thinking in first place why exactly this was added. Any idea how i should check it? Is there any source why this was key check was implemented in first place?

Reply:
Memcached - https://github.com/memcached/memcached/blob/master/doc/protocol.txt#L41

Reply:
![image](https://user-images.githubusercontent.com/3289702/88331669-3c74e800-cd25-11ea-9ad7-824fe2375729.png)

File can be created in Linux.

Reply:
Redis - https://redis.io/topics/data-types-intro#redis-keys

Reply:
Seems like most of them are fine. Will check a windows.

Reply:
Well actually windows doesn't allow it.

Reply:
I decided to revert allowed characters change, i will just str_replace them to `-` and and use `_` for method separator.

Reply:
I will actuallty remove caching stuff from here, results are pretty bad, reflection api is fast enough to handle this on PHP 7.